File: langgraph-studio/src/types/intervention.ts
import type{components}from'./api-generated'

export type InterventionSchema=components['schemas']['InterventionSchema']

export type InterventionPriority='normal'|'urgent'

export type InterventionTarget='all'|'specific'

export type InterventionStatus='pending'|'delivered'|'acknowledged'|'processed'|'waiting_response'

export interface InterventionResponse{
 sender:'agent'|'operator'
 agentId:string|null
 message:string
 createdAt:string
}

export interface Intervention{
 id:string
 projectId:string
 targetType:InterventionTarget
 targetAgentId:string|null
 priority:InterventionPriority
 message:string
 attachedFileIds:string[]
 status:InterventionStatus
 responses:InterventionResponse[]
 createdAt:string
 deliveredAt:string|null
 acknowledgedAt:string|null
 processedAt:string|null
}

export interface CreateInterventionInput{
 projectId:string
 targetType:InterventionTarget
 targetAgentId?:string
 priority:InterventionPriority
 message:string
 attachedFileIds?:string[]
}

export interface InterventionWithMeta extends Intervention{
 targetAgentName:string|null
 deliveryTimeSeconds:number|null
}

File: backend/schemas/intervention.py
from typing import Optional, List, Dict, Any
from datetime import datetime
from .base import BaseSchema


class InterventionSchema(BaseSchema):
    id: str
    project_id: str
    target_type: str
    target_agent_id: Optional[str] = None
    priority: str
    message: str
    attached_file_ids: List[str] = []
    status: str
    acknowledged_at: Optional[datetime] = None
    processed_at: Optional[datetime] = None
    response: Optional[str] = None
    activation_result: Optional[Dict[str, Any]] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
